pytorch_lightning/core/hooks.py

```diff
@@ -115,6 +115,42 @@ def on_epoch_end(self) -> None:
         """
         # do something when the epoch ends
 
+    def on_train_epoch_start(self) -> None:
```

Comment:
@PyTorchLightning/core-contributors any opinion on the name?
I would see the meaning as a hierarchy so `on_epoch_train_start` to evoke that single epoch has more phases...
cc: @awaelchli @justusschock

Reply:
yes... but the hierarchi is based on the loop

Reply:
on_loop_idx1_idx2_idx3